### PR TITLE
Implement A complete App

### PR DESCRIPTION
**Pull request by desorgui**
## Pull request changes

- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (catagories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

### CODE REVIEW

File: config/initializers/devise.rb [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+  # config.parent_controller = 'DeviseController'
+
+  # ==> Mailer Configuration
+  # Configure the e-mail address which will be shown in Devise::Mailer,
+  # note that it will be overwritten if you use your own mailer class
+  # with default "from" parameter.
+  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
+
+  # Configure the class responsible to send e-mails.
+  # config.mailer = 'Devise::Mailer'
```
Comment (omar-muhamad) [inline config/initializers/devise.rb:30]: - [ ] When I try to sign up I didn't get any confirmation email So I can't log in to the app, So please fix that.

image

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @desorgui,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=Once%20you%20have%20finished%20the%20development%20of%20the%20project%2C%20you%20should%20record%20a%20video%20presenting%20the%20features%20of%20the%20project%20you%20built.%20It%20is%20a%20video%20with%20a%20maximum%20length%20of%205%20minutes.%20The%20content%20of%20the%20video%20should%20include%3A), Please add video presentation to your app and provide the link in the Pull Request summary or the README file.

- [ ] As per this [requirement,](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=The%20project%20should%20be%20deployed%20and%20accessible%20online.) Please deploy the application using Heroku(Recommended), Netlify or GitHub Pages.

- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=The%20Creative%20Commons%20license%20of%20the%20design%20requires%20that%20you%20give%20appropriate%20credit%20to%20the%20author.%20Therefore%2C%20you%20must%20do%20it%20in%20the%20README%20of%20your%20project.), Please give appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/).

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: config/initializers/devise.rb (hunk 1 above)
Comment (desorgui) [reply config/initializers/devise.rb:30]: @Omar-Muhamad Thank for your review. Please check your terminal After registration to find the confirmation link

Comment (omar-muhamad) [reply config/initializers/devise.rb:30]: It's supposed to be sent to the registered email or you can use a gem to make the content of the confirmation email opened on a new tab like mailboxer gem.

Comment (desorgui) [reply config/initializers/devise.rb:30]: @Omar-Muhamad, thank you again for your suggestions. I'll implement it.

File: README.md [hunk 2]
```diff
@@ -84,6 +84,13 @@ $ rails s
 - Twitter: [@DGuishny](https://twitter.com/DGuishny)
 - LinkedIn: [Guishny Desor](https://www.linkedin.com/in/guishny-desor-5421a01a9/)
```
Comment (rloterh) [inline README.md:86]: - [ ] OPTIONAL
Kindly deploys the application using Heroku, Netlify or GitHub Pages, and ensure the link to the deployed project is present in the README.

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Desor,  

Your project is complete! There is nothing else to say other than... it's time to merge it 🍾🍾   Congratulations!  🎉💯🌟 

## Highlights 👍
- Correctly deployed application
- Added required models, controllers, and corresponding views
- Devise correctly implemented for authentication

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**